src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -92,22 +95,22 @@ public class FailoverConnectionPlugin implements IConnectionPlugin {
   static final String METHOD_CLOSE = "close";
   static final String METHOD_IS_CLOSED = "isClosed";
 
-  private static final List<String> METHODS_REQUIRE_UPDATED_TOPOLOGY = new ArrayList<>(Arrays.asList(
-      METHOD_COMMIT,
-      "connect",
-      "isValid",
-      "rollback",
-      "setAutoCommit",
-      "setReadOnly",
-      "execute",
-      "executeBatch",
-      "executeLargeBatch",
-      "executeLargeUpdate",
-      "executeQuery",
-      "executeUpdate",
-      "executeWithFlags",
-      "getParameterMetaData"
-  ));
+  private final static Set<String> METHODS_REQUIRE_UPDATED_TOPOLOGY = new ConcurrentHashMap<String, String>() {{
```

Comment:
will you use any Map functions on this Map? Or just searches on the keySet?

Reply:
ConcurrentHashMap.newKeySet()